enterprise_subsidy/apps/content_metadata/api.py

```diff
@@ -124,21 +124,52 @@ def get_geag_variant_id_for_content(self, content_identifier, content_data):
                 variant_id = additional_metadata.get('variant_id')
         return variant_id
 
+    def enroll_by_date_for_content(self, content_data, content_mode):
+        """
+        Determines the enrollment deadline for the given ``content_data``,
+        which could be either a course or course run, hence the branching
+        on content type below.
+
+        Args:
+            content_data: A dictionary of metadata pertaining to a course or course run.
+            content_mode: The (already computed) course mode for the content.
+
+        Returns:
+            A datetime string, or possibly null.
+        """
+        if content_data.get('content_type') == 'course':
+            return course_metadata.get('normalized_metadata', {}).get('enroll_by_date')
```

Comment:
[related note] Similar to below comment about `upgrade_deadline_override`, it doesn't seem that `enroll_by_date` serialized in `normalized_metadata` within enterprise-catalog takes into account `upgrade_deadline_override` ([source](https://github.com/openedx/enterprise-catalog/blob/882034fe51e1f3d39f760232532b6283234c7f43/enterprise_catalog/apps/catalog/serializers.py#L137-L139)).